### PR TITLE
MS-51: profiles-api-rest

### PR DESCRIPTION
**Pull request by ValentinaCollareta**
* Add new model
* Add endpoints + controllers using GET, POST, PUT & DELETE
* Import Postman collection

### CODE REVIEW

File: src/controllers/profiles.js [hunk 1]
```diff
@@ -1,57 +1,74 @@
...
+    });
 };
 
-// Create Profile
 const create = (req, res) => {
-  if (!req.query.id || !req.query.name || !req.query.description) {
-    return res.status(400).send("Some parameters are missing");
+  if (!req.body.name || !req.body.description) {
+    return res.status(400).json({ msg: "Some parameters are missing" });
   }
```
Comment (BaltazarCortina) [inline src/controllers/profiles.js:27]: @ValentinaCollareta this is already being validated in validations/profiles.js, it doesn't need to be validated again here

File: src/routes/profiles.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const profiles = require("../controllers/profiles");
+const validations = require("../validations/profiles");
+const router = require("express").Router();
+
+router.get("/", profiles.getAll);
+router.get("/:id", profiles.getById);
+router.put("/:id", validations.validate, profiles.update);
+router.post("/", validations.validate, validations.required, profiles.create);
```
Comment (BaltazarCortina) [inline src/routes/profiles.js:8]: @ValentinaCollareta shouldn't we check that both parameters are sent (.required) before checking that they meet the other restrictions (.validate)?

**Review by BaltazarCortina — CHANGES_REQUESTED** [review]

File: src/validations/profiles.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+const validate = (req, res, next) => {
+  const name = req.body.name;
+  const description = req.body.description;
+  if (name) {
+    if (name.length > 50) {
+      return res.status(400).send("Name must have less than 50 characters");
+    } else if (typeof name !== "string") {
+      return res.status(400).send("Name must be string");
+    }
```
Comment (BaltazarCortina) [inline src/validations/profiles.js:19]: @ValentinaCollareta we're missing a `next();` after this line

**Review by BaltazarCortina — CHANGES_REQUESTED** [review]

File: src/models/profiles.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const ProfilesSchema = new Schema({
+  name: {
+    type: String,
+    maxLength: 50,
+    required: true,
```
Comment (nicolasblasco) [inline src/models/profiles.js:8]: @ValentinaCollareta we should add the type unique so the name of the profile can't be repeated. You define it like this --> unique: true

File: src/controllers/profiles.js [hunk 5]
```diff
@@ -1,57 +1,71 @@
...
-  }
   const newProfile = {
-    id: (profiles.length + 1).toString(),
-    name: req.query.name,
-    description: req.query.description,
+    name: req.body.name,
+    description: req.body.description,
   };
+  Profiles.create(newProfile);
   return res.status(201).json(newProfile);
```
Comment (nicolasblasco) [inline src/controllers/profiles.js:30]: @ValentinaCollareta we should also return an error response. We could use the same promise that was used in getAll & getById function.

File: src/controllers/profiles.js [hunk 6]
```diff
@@ -1,57 +1,76 @@
...
-    id: (profiles.length + 1).toString(),
-    name: req.query.name,
-    description: req.query.description,
+    name: req.body.name,
+    description: req.body.description,
   };
-  return res.status(201).json(newProfile);
+  Profiles.create(newProfile)
+    .then((newProfile) => {
+      return res.status(200).json(newProfile);
```
Comment (nicolasblasco) [inline src/controllers/profiles.js:31]: @ValentinaCollareta we should send a 201 HTTP status request code

**Review by BaltazarCortina — APPROVED** [review]

**Pull request merged**